src/custom/components/Tokens/TokensTable.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useMemo, useCallback, useEffect } from 'react'
+import { Token } from '@uniswap/sdk-core'
+import { ThemedText } from 'theme'
+import Loader from 'components/Loader'
+import LoadingRows from 'components/LoadingRows'
+import { AutoColumn } from 'components/Column'
+import TokensTableRow from './TokensTableRow'
+import { Label, Wrapper, TableHeader, TableBody, Break, PageButtons, Arrow, ArrowButton, ClickableText } from './styled'
+
+const MAX_ITEMS = 10
+
+const SORT_FIELD = {
+  name: 'name',
+}
+
+type TokenTableParams = {
+  tokensData: Token[] | undefined
+  maxItems?: number
+}
+
+export default function TokenTable({ tokensData, maxItems = MAX_ITEMS }: TokenTableParams) {
+  // sorting
+  const [sortField, setSortField] = useState(SORT_FIELD.name)
+  const [sortDirection, setSortDirection] = useState<boolean>(false)
+
+  // pagination
+  const [page, setPage] = useState(1)
+  const [maxPage, setMaxPage] = useState(1)
+
+  const sortedTokens = useMemo(() => {
+    return tokensData
+      ? tokensData
+          .filter((x) => !!x)
+          .sort((a, b) => {
+            const sortA = a[sortField as keyof Token]
```

Comment:
Why not making `sortField` already typed as `keyof Token` so u don't need to forze it like this?

Reply:
If I set `useState<keyof Token>(SORT_FIELD.name)` then I get this error `Argument of type 'string' is not assignable to parameter of type 'keyof Token | (() => keyof Token)'.`